utils/test_summary.py

```diff
@@ -1,181 +1,114 @@
-#!/usr/bin/env python3
 """
-Test Summary Utility
+    Test Summary Utility
+    This script reads pytest JSON report and displays test results in a tabular format.
+    Useful for Jenkins and CI/CD pipelines to get a clear overview of test results.
 
-This script reads pytest JSON report and displays test results in a tabular format.
-Useful for Jenkins and CI/CD pipelines to get a clear overview of test results.
+    Usage:
+        python utils/test_summary.py [json_report_file]
 
-Usage:
-    python utils/test_summary.py [json_report_file]
-
-Example:
-    python utils/test_summary.py test-results/report.json
+    Example:
+        python utils/test_summary.py test-results/report.json
 """
 
 import json
 import sys
+from collections import defaultdict
 from tabulate import tabulate
-from pathlib import Path
-
 
-def print_test_summary(json_file='test-results/report.json'):
-    """
-    Print test results summary in tabular format
 
-    Args:
-        json_file (str): Path to the pytest JSON report file
-    """
+def print_summary(json_file="test-results/report.json"):
+    """Load pytest JSON report and print summarized test results per spec."""
+    # --- Load JSON report ---
     try:
-        with open(json_file, 'r') as f:
+        with open(json_file, "r") as f:
             data = json.load(f)
-    except FileNotFoundError:
-        print(f"Error: JSON report file not found: {json_file}")
-        print(f"Make sure to run pytest with --json-report --json-report-file={json_file}")
-        sys.exit(1)
-    except json.JSONDecodeError:
-        print(f"Error: Invalid JSON in report file: {json_file}")
+    # Catching specific exceptions helps CI/CD logs surface the real issue.
+    except (IOError, OSError, json.JSONDecodeError) as e:
+        print(f"❌ Error reading report file: {e}")
```

Comment:
Can we avoid the usage of emojis here? They broke the windows pipeline in jenkins. That is a problem in the jenkins runner but for now we can avoid using them to ensure stability